### PR TITLE
[BOOST-5112] feat(evm): add pegged variable criteria incentive

### PR DESCRIPTION
**Pull request by Quazia**
🚨 Please review the [guidelines for contributing](https://github.com/boostxyz/boost-protocol/blob/main/.github/CONTRIBUTING.md) to this repository.

### Description
Adds the contracts and SDK files for pegged variable criteria incentives.
💔 Thank you!

### CODE REVIEW

**Comment by sammccord** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Are you sure you want to be submitting a change without including a changeset? If you're just changing docs or tests, you probably don't need to. See [the publishing section of the README](https://github.com/boostxyz/boost-protocol?tab=readme-ov-file#changesets--publishing) for more info.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 4ca147ac90a40ca90a9bb6ed4ecb3c881b4b5d2c
</p>

**Comment by sammccord** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Are you sure you want to be submitting a change without including a changeset? If you're just changing docs or tests, you probably don't need to. See [the publishing section of the README](https://github.com/boostxyz/boost-protocol?tab=readme-ov-file#changesets--publishing) for more info.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 3858ecd04d9a54830ce924e5e8ef35094f6b793d
</p>

File: packages/sdk/src/Incentives/ERC20PeggedVariableCriteriaIncentive.ts [hunk 1]
```diff
@@ -0,0 +1,563 @@
...
+} from '../utils';
+import type { IncentiveCriteria } from './ERC20VariableCriteriaIncentive';
+
+export { erc20PeggedVariableCriteriaIncentiveAbi };
+
+/**
+ * The object representation of a `Erc20PeggedVariableCriteriaIncentive.InitPayload`
+ *
+ * @export
+ * @interface E20PeggedVariableCriteriaIncentive
```
Comment (mmackz) [inline packages/sdk/src/Incentives/ERC20PeggedVariableCriteriaIncentive.ts:50]: typo

Comment (mmackz) [inline packages/sdk/src/Incentives/ERC20PeggedVariableCriteriaIncentive.ts:50]:
```suggestion
 * @interface ERC20PeggedVariableCriteriaIncentive
```

File: packages/sdk/src/Incentives/ERC20PeggedVariableCriteriaIncentive.ts [hunk 2]
```diff
@@ -0,0 +1,563 @@
...
+        address: this.assertValidAddress(),
+        args: [],
+        // biome-ignore lint/suspicious/noExplicitAny: Accept any shape of valid wagmi/viem parameters, wagmi does the same thing internally
+        ...(params as any),
+      },
+    );
+  }
+
+  /**
+   * Fetches the IncentiveCriteria struct from the contract
```
Comment (mmackz) [inline packages/sdk/src/Incentives/ERC20PeggedVariableCriteriaIncentive.ts:179]: /**
   * Retrieves the maximum reward amount per claim from the contract.
   *
   * @param {?ReadParams} [params] Optional parameters for the read operation
   * @returns {Promise<bigint>} The maximum reward amount per claim
   */

File: packages/sdk/src/Incentives/ERC20PeggedVariableCriteriaIncentive.ts [hunk 3]
```diff
@@ -0,0 +1,563 @@
...
+
+  /**
+   * The reward amount issued for each claim
+   *
+   * @public
+   * @async
+   * @param {?ReadParams} [params]
+   * @returns {Promise<bigint>}
+   */
+  public async getReward(params?: ReadParams) {
```
Comment (mmackz) [inline packages/sdk/src/Incentives/ERC20PeggedVariableCriteriaIncentive.ts:274]: These methods 
- getReward
- getLimit

are named just limit and reward in other incentive types. We should keep it consistent.

**Review by mmackz — APPROVED** [review]
Left some comments but otherwise LGTM

**Comment by sammccord** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Are you sure you want to be submitting a change without including a changeset? If you're just changing docs or tests, you probably don't need to. See [the publishing section of the README](https://github.com/boostxyz/boost-protocol?tab=readme-ov-file#changesets--publishing) for more info.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against a7fa0a25a6feaddd95fbc00b5043a0067b95b85a
</p>

File: packages/sdk/src/Incentives/ERC20PeggedVariableCriteriaIncentive.ts (hunk 3 above)
Comment (Quazia) [reply packages/sdk/src/Incentives/ERC20PeggedVariableCriteriaIncentive.ts:275]: The issue is that it's causing a collision with other params in the file but if it works ship it

**Comment by sammccord** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Are you sure you want to be submitting a change without including a changeset? If you're just changing docs or tests, you probably don't need to. See [the publishing section of the README](https://github.com/boostxyz/boost-protocol?tab=readme-ov-file#changesets--publishing) for more info.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against d75c67db1ce46a05f3f7c4252e65683a197a9d87
</p>

**Comment by sammccord** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Are you sure you want to be submitting a change without including a changeset? If you're just changing docs or tests, you probably don't need to. See [the publishing section of the README](https://github.com/boostxyz/boost-protocol?tab=readme-ov-file#changesets--publishing) for more info.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 2946ae9aa35c9403fcdd1b7362ae864d535aa0ff
</p>

**Comment by sammccord** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Are you sure you want to be submitting a change without including a changeset? If you're just changing docs or tests, you probably don't need to. See [the publishing section of the README](https://github.com/boostxyz/boost-protocol?tab=readme-ov-file#changesets--publishing) for more info.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 26ea75b18d29f2de073d8063fb78cc510c4762d7
</p>

**Comment by mmackz** [comment]
Need to merge #329 first

**Pull request merged**